cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/FileStreamAdmin.java

```diff
@@ -410,82 +410,91 @@ public StreamConfig create(final StreamId streamId, @Nullable final Properties p
     // User should have write access to the namespace
     NamespaceId streamNamespace = streamId.getParent();
     ensureAccess(streamNamespace, Action.WRITE);
-    // revoke privileges to make sure there is no orphaned privileges
-    privilegesManager.revoke(streamId);
+
     final Properties properties = (props == null) ? new Properties() : props;
+    String ownerPrincipal = properties.containsKey(Constants.Security.PRINCIPAL) ?
+      properties.getProperty(Constants.Security.PRINCIPAL) : null;
+
+    if (exists(streamId)) {
+      // if stream exists then make sure owner for this create request is same
+      verifyOwner(streamId, ownerPrincipal);
+      // stream create is an idempotent operation as of now so just return null and don't do anything
+      return null;
```

Comment:
It seems like the stream creation is idempotent. See this line which I removed: https://github.com/caskdata/cdap/pull/8126/files#diff-f8d2839dec129f7a807966f93b41d7faL446
The  java doc of api says so too although it doesn't mark it as Nullable.
https://github.com/caskdata/cdap/blob/release/4.1/cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/StreamAdmin.java#L104

I don't know why it was made idempotent. I didn't change it in this PR as it's a backward incompatible change and I wanted to discuss the correct behavior.